src/main/java/digital/slovensko/autogram/core/eforms/EFormUtils.java

```diff
@@ -493,27 +493,19 @@ public static ManifestXsltEntry selectXslt(ArrayList<ManifestXsltEntry> entries,
         if (xsltTarget != null)
             entries.removeIf(entry -> !xsltTarget.equals(entry.target()));
 
-        if (entries.size() == 1)
-            return entries.get(0);
+        filterIfExist(entries, e -> e.mediaDestination().equals("sign"));
+        filterIfExist(entries, e -> List.of("HTML", "XHTML").contains(e.destinationType()));
+        filterIfExist(entries, e -> e.mediaDestination().equals("view"));
+        filterIfExist(entries, e -> e.destinationType().equals("XHTML"));
 
-        if (entries.stream().filter(entry -> entry.mediaDesination().equals("sign")).count() > 0)
-            entries.removeIf(entry -> !entry.mediaDesination().equals("sign"));
+        filterIfExist(entries, e -> e.language().equals("sk"));
+        filterIfExist(entries, e -> e.language().equals("en"));
 
-        if (entries.stream().filter(entry -> entry.destinationType().equals("XHTML")).count() > 0)
-            entries.removeIf(entry -> !entry.destinationType().equals("XHTML"));
-
-        else if (entries.stream().filter(entry -> entry.destinationType().equals("HTML")).count() > 0)
-            entries.removeIf(entry -> !entry.destinationType().equals("HTML"));
-
-        else if (entries.stream().filter(entry -> entry.destinationType().equals("TXT")).count() > 0)
-            entries.removeIf(entry -> !entry.destinationType().equals("TXT"));
-
-        if (entries.stream().filter(entry -> entry.language().equals("sk")).count() > 0)
-            entries.removeIf(entry -> !entry.language().equals("sk"));
-
-        else if (entries.stream().filter(entry -> entry.language().equals("en")).count() > 0)
-            entries.removeIf(entry -> !entry.language().equals("en"));
+        return entries.stream().findFirst().orElse(null);
+    }
```

Comment:
Toto pre mna je trosku zle citatelne kedze ta metoda "mutuje" entries. Predtym som videl, ze najprv sa hlada `sign` potom xhtml potom html a potom txt. Tu to nevidim.

Reply:
mne tam vadilo, že sa príliš veľa opakovalo. Rozumiem, že metódy, ktoré menia parametre sú nešťastné. Ako kompromis by som metóde `filterIfExist` dal return type entries a nebude meniť tie entries v parametri ale ich kópiu.